src/deviceCommunication/commUSB.cpp

```diff
@@ -0,0 +1,90 @@
+/******************************************************************************
+ * Copyright (C) 2022 by Gschwind, Weber, Schoch, Niederberger                *
+ *                                                                            *
+ * This file is part of linescaleGUI.                                         *
+ *                                                                            *
+ * LinescaleGUI is free software: you can redistribute it and/or modify       *
+ * it under the terms of the GNU General Public License as published by       *
+ * the Free Software Foundation, either version 3 of the License, or          *
+ * (at your option) any later version.                                        *
+ *                                                                            *
+ * LinescaleGUI is distributed in the hope that it will be useful,            *
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of             *
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the               *
+ * GNU General Public License for more details.                               *
+ *                                                                            *
+ * You should have received a copy of the GNU General Public License          *
+ * along with linescaleGUI. If not, see <http://www.gnu.org/licenses/>.       *
+ ******************************************************************************/
+/**
+ * @file commUSB.cpp
+ * @authors Gschwind, Weber, Schoch, Niederberger
+ *
+ *
+ */
+
+#include "commUSB.h"
+
+CommUSB::CommUSB(deviceInfo identifier) {
+    this->identifier = identifier;
+};
+
+CommUSB::~CommUSB() {
+    CommUSB::disconnectDevice();
+    delete serialPort;
+}
+
+void CommUSB::disconnectDevice() {
+    if(serialPort != nullptr)
+    {
+        serialPort->close();
+        connected = false;
+        emit changedStateDevice(connected);
+        delete serialPort;
+        serialPort = nullptr;
+    }
```

Comment:
Manual memory management is very error-prone. E.g. if you forget to set `serialPort` to `nullptr` a double-free could occurr.

To prevent such mistakes [`std::unique_ptr](https://en.cppreference.com/w/cpp/memory/unique_ptr) can be used. It manages allocation/construction and deallocation/destruction of heap memory.
E.g:
```C++
serialPort = std::make_unique<QSerialPort>(); // to construct
serialPort.reset(); // to destruct
```
Or, remove the indirection:
```C++
QSerialPort serialPort; // in member variable declaration
// ....
serialPort.close(). // to close it, without deleting
serialPort.open(...). // to open without construction
```

Or, set this class as its parent (then it will be automatically destroyed once this instance is):
```C++
serialPort = new QSerialPort(this); // in constructor
// ....
serialPort.close(). // to close it, without deleting
serialPort.open(...). // to open without construction
```

Reply:
Removed data on heap, changed to `QSerialPort serialPort;`